jmx/src/test/java/org/jboss/as/jmx/PluggableMBeanServerTestCase.java

```diff
@@ -165,6 +199,10 @@ public void preDeregister() throws Exception {
         public void postDeregister() {
         }
 
+        @Override
+        public void proceed() {
+            sendNotification(new Notification("testtccl", 1, 1));
```

Comment:
If you change the return type of this method to ClassLoader, then you can add this:

```
return WildFlySecurityManager.getCurrentContextClassLoaderPrivileged();
```

And then the test can validate that it returns newOutTccl. That both serves as a sanity check that newOutTccl was in place before the notification handling, and confirms that it got restored.

Reply:
@bstansberry  I've Renamed it to testNotificationTccl. Initailly I did the same as returntype ClassLoader, But I thought this is not invoking notification was just a test,  so I wrote it inside notification, In that way it actually invoked notification and test the same. Is that okay to have this or still you want me to change?

Reply:
@bstansberry  Initailly I did the same as returntype ClassLoader, But I thought this is not invoking notification was just a test, so I wrote it inside notification, In that way it actually invoked notification and test the same. Is that okay to have this or still you want me to change?

Reply:
What your NotificationListener is doing is fine -- it checks that the expected TCCL is in place when it is invoked. I was suggesting an additional check that the expected TCCL (newOutTccl) is in place after the notification call is done.

But this is ok as is.